src/Temporalio/Worker/Tuning/WorkerTuner.cs

```diff
@@ -0,0 +1,106 @@
+namespace Temporalio.Worker.Tuning
+{
+    /// <summary>
+    /// WorkerTuners allow for the dynamic customization of some aspects of worker configuration.
+    /// </summary>
+    public abstract class WorkerTuner
```

Comment:
I have an alternative suggestion here. .NET prefers interfaces and the default worker tuner can just be composite, so how about something like:

```csharp
public interface IWorkerTuner
{
    ISlotSupplier WorkerTaskSlotSupplier { get; }
    ISlotSupplier ActivityTaskSlotSupplier { get; }
    ISlotSupplier WorkflowTaskSlotSupplier { get; }
}

public record WorkerTuner(
    ISlotSupplier WorkerTaskSlotSupplier,
    ISlotSupplier ActivityTaskSlotSupplier,
    ISlotSupplier WorkflowTaskSlotSupplier) : IWorkerTuner
{
    public static WorkerTuner CreateResourceBased(...)

    public static WorkerTuner CreateFixed(...)
}
```

And get rid of `CompositeTuner` and accept `IWorkerTuner` in the worker options. I think this is more idiomatic .NET (have interface and common impl as the primary impl). Arguably it'd be more idiomatic in .NET (unlike Python and others) to also make `ResourceBasedTuner` a class with a constructor instead of a static helper here. So could be something like:

```csharp
public interface IWorkerTuner
{
    ISlotSupplier WorkerTaskSlotSupplier { get; }
    ISlotSupplier ActivityTaskSlotSupplier { get; }
    ISlotSupplier WorkflowTaskSlotSupplier { get; }
}

public record WorkerTuner(
    ISlotSupplier WorkerTaskSlotSupplier,
    ISlotSupplier ActivityTaskSlotSupplier,
    ISlotSupplier WorkflowTaskSlotSupplier) : IWorkerTuner;

public class ResourceBasedTuner : WorkerTuner
{
    public ResourceBasedTuner(
        double targetMemoryUsage,
        double targetCpuUsage,
        ResourceBasedSlotSupplierOptions? workflowOptions = null,
        ResourceBasedSlotSupplierOptions? activityOptions = null,
        ResourceBasedSlotSupplierOptions? localActivityOptions = null)
        : base(
            new ResourceBasedSlotSupplier(workflowOptions ?? new(), new(targetMemoryUsage, targetCpuUsage),
            new ResourceBasedSlotSupplier(activityOptions ?? new(), new(targetMemoryUsage, targetCpuUsage),
            new ResourceBasedSlotSupplier(localActivityOptions ?? new(), new(targetMemoryUsage, targetCpuUsage))
    {
    }
}

public class FixedSizeTuner : WorkerTuner
{
    public FixedSizeTuner(int workflowTaskSlots, int activityTaskSlots, int localActivitySlots)
        : base(
            new FixedSizeSlotSupplier(workflowTaskSlots),
            new FixedSizeSlotSupplier(activityTaskSlots),
            new FixedSizeSlotSupplier(localActivitySlots))
    {
    }
}
```

I think that last one is the most idiomatic